# 数据集预处理

以数据集为cvs文件为例,使用pandas进行数据集的预处理,本文以kaggle的房价预测数据集为例,已有的数据集如下：

- data_description.txt 
- sample_submission.csv 
- test.csv 
- train.csv

[数据集链接](https://www.kaggle.com/competitions/house-prices-advanced-regression-techniques/data)

# 1.数据集加载

In [3]:
import pandas as pd

#导入数据
test_data = pd.read_csv("E:/PythonProject/Kaggle/Data/housePrise/test.csv")
train_data = pd.read_csv("E:/PythonProject/Kaggle/Data/housePrise/train.csv")
#查看是否导入成功
print(train_data.head(5))

   Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD   

In [4]:
#查看数据信息
train_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

 “79  SaleCondition  1460 non-null   object”从结果可以看出，train_data数据有81列，对应各种特征因子，如房价，房型这些，有1460行，每行的缺省值数目也写出来了，进而可以快速筛选缺省值，如大量空条目的列：PoolQC、MiscFeature等；同时也可以看到列的数值对应3种数据类型，分别是float64(3), int64(35), object(43)
